modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryManager.java

```diff
@@ -1203,7 +1219,7 @@ protected void runQuery(GridCacheQueryInfo qryInfo) {
                         break;
                     }
 
-                    if (type == SCAN || type == INDEX)
```

Comment:
This change might affect IndexQuery performance due to useless generics handling. Let's check how ScanQuery record this event.

Reply:
Recording of Index query object reading events moved to `IndexQueryProcessor#queryLocal`.